app/controllers/labware_controller.rb

```diff
@@ -179,8 +160,8 @@ def lab_view
 
   def labware_params
     permitted = %i[volume concentration]
-    permitted << :name if current_user.administrator?
-    permitted << :plate_purpose_id if current_user.administrator? || current_user.lab_manager?
+    permitted << :name if can? :rename, Labware
+    permitted << :plate_purpose_id if can? :change_purpose, Labware
     params.require(:labware).permit(permitted)
```

Comment:
CanCanCan actually provides its own way of managing permitted attributes. But I figured that can wait for another day.